### PR TITLE
Integrate GZCoav plugin + models into GZSitl

### PR DESCRIPTION
**Pull request by guiccbr**
Please see commit messages for more details

### CODE REVIEW

File: gzsitl/gzsitl_visflags_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#include "gzsitl_visflags_plugin.hh"
```
Comment (rchiossi) [inline gzsitl/gzsitl_visflags_plugin.cc:17]: Missing copyright?

File: gzsitl/gzsitl_plugin.cc [hunk 2]
```diff
@@ -17,6 +17,10 @@
 #include "defines.hh"
 #include "gzsitl_plugin.hh"
 
+#define TARGET_POSE_PUB_FREQ_HZ 50
+#define VEHICLE_POSE_PUB_FREQ_HZ 50
+#define SUBS_TARGET_POSE_SUB_MAX_RESPOND_TIME 1000
```
Comment (rchiossi) [inline gzsitl/gzsitl_plugin.cc]: response instead of respond maybe?

Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc]: It would be nice to at least add a prefix on these macros

Comment (guiccbr) [inline gzsitl/gzsitl_plugin.cc]: @dakerfp, can you give an example?

File: gzsitl/gzsitl_plugin.cc [hunk 3]
```diff
@@ -276,14 +329,44 @@ void GZSitlPlugin::set_global_pos_coord_system(
...
+
+math::Pose GZSitlPlugin::get_subs_target_pose()
 {
-    pose->Set(local_position.x,
-              -local_position.y,
-              -local_position.z,
-              attitude.roll,
-              -attitude.pitch,
-              -attitude.yaw);
+    subs_target_pose_mtx.lock();
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc]: use lock_guard

File: gzsitl/gzsitl_plugin.cc [hunk 4]
```diff
@@ -276,14 +329,44 @@ void GZSitlPlugin::set_global_pos_coord_system(
...
+    return pose_copy;
+}
+
+bool GZSitlPlugin::is_target_overridden()
+{
+    using namespace std::chrono;
+
+    time_point<system_clock> curr_time = system_clock::now();
+
+    if (duration_cast<milliseconds>(curr_time - subs_target_pose_sub_recv_time)
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc]: simply return duration_cast<milliseconds>(curr_time - subs_target_pose_sub_recv_time)
-            .count() < SUBS_TARGET_POSE_SUB_MAX_RESPOND_TIME

**Comment by guiccbr** [comment]
@dakerfp and @rchiossi, please see the new commits with the implementation of your suggestions.

**Comment by rchiossi** [comment]
I'm not sure if you still want to add more patches on top of those 4, but those look good to me.

**Comment by dakerfp** [comment]
LGTM

File: gzsitl/CMakeLists.txt [hunk 5]
```diff
@@ -17,5 +17,5 @@ add_library(gzsitl_visflags_plugin SHARED gzsitl_visflags_plugin.cc)
 target_link_libraries(gzsitl_plugin ${GAZEBO_libraries} ${LDFLAGS})
 target_link_libraries(gzsitl_visflags_plugin ${GAZEBO_libraries} ${LDFLAGS})
 
-# Set Gazebo Environment Variables
-install(TARGETS gzsitl_plugin gzsitl_visflags_plugin DESTINATION ${GZSITL_PLUGIN_PATH})
+install(TARGETS gzsitl_plugin gzsitl_visflags_plugin
```
Comment (dakerfp) [inline gzsitl/CMakeLists.txt]: Put each element in its own row.

**Pull request closed without merging**